src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -250,6 +251,8 @@ InterpreterSelectQuery::InterpreterSelectQuery(
         source_header = input_pipe->getHeader();
     }
 
+    if (context->getSettingsRef().with_global)
+        ApplyWithAliasVisitor().visit(query_ptr);
```

Comment:
I think it need to be moved into `TreeRewriter`. Probably,  `TreeRewriter::normalize`

Reply:
It's somehow consistent with the current CTE implementation ...